shared/js/otSdkDriver.js

```diff
@@ -1001,20 +1001,28 @@ loop.OTSdkDriver = (function() {
         // Nothing to do for the success case.
         return;
       }
-      if (!(error.message && error.message === "DENIED")) {
-        // We free up the publisher here in case the store wants to try
-        // grabbing the media again.
-        if (this.publisher) {
-          this.publisher.off("accessAllowed accessDenied accessDialogOpened streamCreated");
-          this.publisher.destroy();
-          delete this.publisher;
-          delete this._mockPublisherEl;
-        }
-        this.dispatcher.dispatch(new sharedActions.ConnectionFailure({
-          reason: FAILURE_DETAILS.UNABLE_TO_PUBLISH_MEDIA
-        }));
-        this._notifyMetricsEvent("sdk.exception." + error.code + "." + error.message);
+
+      if (error.message && error.message === "DENIED") {
+        // In the DENIED case, this will be handled by _onPublishDenied.
+        return;
       }
+
+      // We free up the publisher here in case the store wants to try
+      // grabbing the media again.
+      if (!this.publisher) {
+        return;
+      }
+
+      this.publisher.off("accessAllowed accessDenied accessDialogOpened streamCreated");
+      this.publisher.destroy();
+      delete this.publisher;
+      delete this._mockPublisherEl;
+
+      this.dispatcher.dispatch(new sharedActions.ConnectionFailure({
+        reason: FAILURE_DETAILS.UNABLE_TO_PUBLISH_MEDIA
```

Comment:
> Before this PR, this would be dispatched even without a publisher, but now it only dispatches it if there's a publisher to freed up. Is that desired?

I think that's fine. When initialising the publisher we get a new object, and we're unlikely to clear it out elsewhere without handling it properly.